plugins/kubernetes-backend/src/service/KubernetesBuilder.ts

```diff
@@ -218,27 +233,100 @@ export class KubernetesBuilder {
   protected buildRouter(
     objectsProvider: KubernetesObjectsProvider,
     clusterSupplier: KubernetesClustersSupplier,
+    catalogClient: CatalogApi,
   ): express.Router {
     const logger = this.env.logger;
     const router = Router();
     router.use(express.json());
 
+    // TODO fix any
+    const getEntityByReq = async (req: Request<any>) => {
+      const rawEntityRef = req.query.entity;
+      if (rawEntityRef && typeof rawEntityRef !== 'string') {
+        throw new InputError(`entity query must be a string`);
+      } else if (!rawEntityRef) {
+        throw new InputError('entity is a required field');
+      }
+      let entityRef: CompoundEntityRef | undefined = undefined;
+
+      try {
+        entityRef = parseEntityRef(rawEntityRef);
+      } catch (error) {
+        throw new InputError(`Invalid entity ref, ${error}`);
+      }
+
+      function getBearerToken(header?: string): string | undefined {
+        return header?.match(/Bearer\s+(\S+)/i)?.[1];
+      }
+      const entity = await catalogClient.getEntityByRef(entityRef, {
+        token: getBearerToken(req.headers.authorization),
+      });
+
+      if (!entity) {
+        throw new InputError(`Entity ref missing, ${entityRef}`);
+      }
+      return entity;
+    };
+
+    // Deprecated, will be removed soon
+    // see https://github.com/backstage/backstage/issues/11309
     router.post('/services/:serviceId', async (req, res) => {
       const serviceId = req.params.serviceId;
       const requestBody: ObjectsByEntityRequest = req.body;
       try {
         const response = await objectsProvider.getKubernetesObjectsByEntity(
-          requestBody,
+          requestBody.entity,
+          requestBody.auth!,
         );
         res.json(response);
       } catch (e) {
+        console.log(e);
         logger.error(
           `action=retrieveObjectsByServiceId service=${serviceId}, error=${e}`,
         );
         res.status(500).json({ error: e.message });
       }
     });
 
+    router.post('/resources/workloads', async (req, res) => {
+      const entity = await getEntityByReq(req);
+
+      try {
+        const response = await objectsProvider.getKubernetesObjectsByEntity(
+          entity,
+          req.body.auth,
+        );
+        res.json(response);
+      } catch (e) {
+        logger.error(
+          `action=/resources/workloads entityRef=${stringifyEntityRef(
+            entity,
+          )}, error=${e}`,
+        );
+        res.status(500).json({ error: e.message });
+      }
+    });
+
+    router.post('/resources/custom', async (req, res) => {
```

Comment:
hmm, yeah I think that makes sense

Reply:
I'm not sure how possible this will be due to the repeated custom resources field needed, it might be possible but I think it would be even more confusing 😞

Reply:
Yeah bit tricky that bit tbh. I wonder if we should make it more explicit with something like `/resources/custom/query` or that kind of thing? I do slightly wonder why they are two different endpoints as well, maybe even a single `/resource/query`?